x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -57,4 +84,26 @@ static long convertFromCalendarToUTC(long value, Calendar cal) {
 
         return convertedDateTime.toInstant().toEpochMilli();
     }
+
+    static String asStringTimestampFromNanos(long nanos) {
+        return asStringTimestampFromNanos(nanos, ZoneId.systemDefault());
+    }
+
+    static String asStringTimestampFromNanos(long nanos, ZoneId zoneId) {
+        ZonedDateTime zdt = ZonedDateTime.ofInstant(Instant.ofEpochMilli(toMilliSeconds(nanos)), zoneId);
+        return StringUtils.toString(zdt.withNano((int) (nanos % 1_000_000_000)));
+    }
+
+    static long randomNanos() {
+        // Return a number which is at least 20:00:00.000000000 to avoid switching to negative values when a UTC-XX hours is applied
+        return randomLongBetween(72000000000000L, Long.MAX_VALUE);
```

Comment:
Since the rest of the many-0s-numbers are having the thousands `_`-separated, might make sense to do it here too.